src/components/license-activation/LicenseActivationPage.jsx

```diff
@@ -1,13 +1,52 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import { Redirect, useParams } from 'react-router-dom';
+import { AppContext } from '@edx/frontend-platform/react';
+import { Alert } from '@edx/paragon';
+import { logInfo } from '@edx/frontend-platform/logging';
 
+import { getAuthenticatedUser } from '@edx/frontend-platform/auth';
+
+import { useRenderContactHelpText } from '../../utils/hooks';
+import { UserSubsidyContext } from '../enterprise-user-subsidy/UserSubsidy';
+import { LICENSE_STATUS } from '../enterprise-user-subsidy/data/constants';
+import LicenseActivationErrorAlert from './LicenseActivationErrorAlert';
 import LicenseActivation from './LicenseActivation';
-import AuthenticatedPage from '../app/AuthenticatedPage';
 
-const LicenseActivationPage = () => (
-  <AuthenticatedPage useEnterpriseConfigCache={false}>
-    <LicenseActivation />
-  </AuthenticatedPage>
+export const LOADING_MESSAGE = 'Your enterprise license is being activated! You will be automatically redirected to your organization\'s learner portal shortly.';
+
+const LicenseActivationPage = () => {
+  const user = getAuthenticatedUser();
+  const { enterpriseConfig } = useContext(AppContext);
+  const { subscriptionLicense } = useContext(UserSubsidyContext);
+  const { activationKey } = useParams();
+  const renderContactHelpText = useRenderContactHelpText(enterpriseConfig);
+
+  if (!subscriptionLicense || subscriptionLicense.status !== LICENSE_STATUS.ASSIGNED) {
+    logInfo(`User ${user.id} has no license to activate.`);
+    return (
+      <Redirect
+        to={{
+          pathname: `/${enterpriseConfig.slug}`,
+        }}
+      />
+    );
+  }
+
+  if (activationKey !== subscriptionLicense.activationKey) {
+    logInfo(
```

Comment:
nit/curious: Is there a chance we call `logInfo` more times than intended here (i.e., anytime the component re-renders with this matching conditional, we'd be calling `logInfo` I believe) even if we only intended to log it once? Should this `logInfo` happen in a `useEffect` so it doesn't occur on component re-renders when the `activationKey` and `subscriptionLicense.activationKey` remains the same between re-renders?